shared/teams/delete-team/index.tsx

```diff
@@ -69,7 +69,7 @@ const ReallyDeleteTeam = (props: Props) => {
       onBack()
     }
   }, [deleteWaiting, prevDeleteWaiting, onBack, error])
-  React.useEffect(() => clearWaiting(), [clearWaiting])
+  React.useEffect(() => () => clearWaiting(), [clearWaiting])
```

Comment:
This needs to be an empty deps array. This function is getting recreated in `mergeProps`. The behavior is wrong this way because other stateProps are changing, triggering this before unmount.